lib/rack/test/cookie_jar.rb

```diff
@@ -32,6 +32,7 @@ def initialize(raw, uri = nil, default_host = DEFAULT_HOST)
 
         if @options['domain']
           @exact_domain_match = false
+          @options['domain'].sub!(/^\./, '')
```

Comment:
If we had to use a regexp, we should probably use `\A` and not `^`.  It looks like the `raw` argument is passed by the user, and I'm not sure there is a check that it isn't multiline.  For simple things like this, a regexp is probably overkill. Maybe:

```ruby
       if domain = @options['domain']
          @exact_domain_match = false
          domain[0] = '' if domain[0] == '.'
```

Reply:
That is a neat trick. Added that.

Reply:
Slightly faster to do the `domain = @options['domain']` in the `if` as well, as that saves two method calls.

Looks like there are failing tests in CI, but that is due to changes in rack itself, not from this pull request.  I'll see if I can fix that tomorrow, as it is getting late here.